scripts/globals/combat/ranged_utilities.lua

```diff
@@ -112,3 +112,20 @@ xi.combat.ranged.accuracyDistancePenalty = function(attacker, defender)
 
     return penalty
 end
+
+xi.combat.ranged.shouldUseAmmo = function(attacker)
+    if attacker:isPC() then
+        local recycleChance = attacker:getMod(xi.mod.RECYCLE) + attacker:getMerit(xi.merit.RECYCLE)
+
+        if attacker:hasStatusEffect(xi.effect.UNLIMITED_SHOT) then
+            attacker:delStatusEffect(xi.effect.UNLIMITED_SHOT)
+            recycleChance = 100
+        end
+
+        if math.random(0, 99) >= recycleChance then
```

Comment:
For consistency with this kind of checks, can we
`math.random(1, 100) <= recycleChance`